src/control/cmd/daos/pool.go

```diff
@@ -265,6 +300,15 @@ func (cmd *poolQueryCmd) Execute(_ []string) error {
 		return err
 	}
 
+	if ranklistPtr != nil {
+		if cmd.ShowEnabledRanks {
+			pqr.EnabledRanks = system.MustCreateRankSet(generateRankSet(ranklist))
```

Comment:
This is fine for now, but just FYI we are working on improvements to the `system.RankSet` type so that we can avoid the intermediate string representation which is inefficient at best. Once that work is landed we can adapt this code to create them directly from the numeric rank lists.

Reply:
I see. Thank you!